### DISCUSSION TITLE
Review requests to allow access to camera and microphone

### DISCUSSION THREAD

**Discussion by lwcorp** [sandboxie-plus/Sandboxie#1942] [Various ideas and suggestions]
As blockage of camera and microphone is possibly one of Sandboxie's most problematic issues, there's feature request #1669 about it.

But most users report it as a bug rather than a feature request.
Can you please **close** their bug reports with a referral to the aforementioned feature request?
Thanks!

1. #518
1. #703
1. #1119
1. #1208
1. #1824
1. #2130
1. #2197
1. #2436
1. #2508
1. #2718
1. #2925

**Comment by isaak654** [comment]
There seems to be a bug according to https://github.com/sandboxie-plus/Sandboxie/issues/1208#issuecomment-1224056646 and https://github.com/sandboxie-plus/Sandboxie/issues/1208#issuecomment-1224526046, but it is currently not clear what decision the lead developer will make.

**Reply by lwcorp** [reply]
Bug or feature request, this is the single most important reason it's hard for me to get people to use Sandboxie.
Why aren't these issues at least merged?

**Reply by isaak654** [reply]
> Bug or feature request, this is the single most important reason it's hard for me to get people to use Sandboxie.

Feel free to hire or invite a developer of your choice into this open source repository.

**Reply by lwcorp** [reply]
I see your point, but for now we can at least merge the requests, right?

**Reply by isaak654** [reply]
For now, I would be more inclined to update this discussion with new related issues.

**Reply by lwcorp** [reply]
Sure, but for the moment the OP lists 7 that can be merged.

**Reply by isaak654** [reply]
I'm sorry, but I disagree... most of those were not even reproduced / reviewed so far.

**Reply by lwcorp** [reply]
But don't they all deal with the same subject?
Well, I guess maybe each of them should at least link to this discussion.

**Reply by DavidXanatos** [reply]
The thing is this issue seams some what individual, i tried https://www.onlinemictest.com/de/ and https://de.webcamtests.com/ in my sandbox and booth worked fine, now a web browser may use a different api as some chat client.

**Reply by lwcorp** [reply]
Which browser / OS worked for you?

**Reply by jedimasterspaz** [reply]
Using https://de.webcamtests.com:
The webcam works in Windows 11 in Edge outside of Sandboxie Classic 5.60.3.
The webcam doesn't work in Windows 11 in Chrome inside of Sandboxie Classic 5.60.3.
![Screenshot 2022-12-03 23 43 22](https://user-images.githubusercontent.com/72579094/205480043-e89f8806-8a0d-44ac-9b61-cfd761163f8a.png)
![Screenshot 2022-12-03 23 43 12](https://user-images.githubusercontent.com/72579094/205480045-d0357d6b-3bdd-4e3c-a00a-46a04b0f81d6.png)

**Comment by Incog5** [comment]
Camera started working with one of the new Sandboxie/Chrome updates, not sure exactly with what version it started working for me.

Mic wasn't working by default, but **how I got the microphone to work:**

https://mictests.com/
Should trigger microphone popup, allow it, and refresh page.
In the drop down, the default microphone selection won't work. (Unsandboxed, the default one works.)
I had two more additional selections, the last one Microphone Array worked.


If the web app allows you to select the microphone type, select the same one as the test.  This was successful with google voice and discord.
The same problem might happen for webcam, can't use default have to select another one, but i haven't confirmed this as my webcam is working.

Edit: corrected link

**Reply by lwcorp** [reply]
I believe you meant https://mictests.com (with https and tests in plural).
I can't get mic to work there in Chrome since I have just a single choice (the default).
But the mic does work natively in Firefox, although it doesn't even let me choose which mic, so it just uses the internal one.

Also like you wrote, webcam does indeed suddenly work in Chrome, for example in https://webcamtests.com
Then again, in Firefox the webcam also works but again without letting me choose which webcam to choose. So it just chooses on its own the internal one and I can't change that.

**Reply by bastik-1001** [reply]
I just tested 1.9.3 on Windows 7 and Firefox 113.0.1 with an old microphone (as part of an old headset, I had not thrown away) which I plugged in. The website for the micetest, shows two inputs, one of which is a digital input that does not do anything, and the microphone I plugged in, I could select the device and it recorded my voice. This works in a hardened box with data protection (red box)

**Reply by DavidXanatos** [reply]
The camera issue is as far as i can tell windows 11 specific and can be remedied using a compartment type box with open COM.
I assume the microphone will be similar: https://github.com/sandboxie-plus/Sandboxie/issues/1669#issuecomment-1556199041

**Reply by lwcorp** [reply]
> Also like you wrote, webcam does indeed suddenly work in Chrome, for example in https://webcamtests.com Then again, in Firefox the webcam also works but again without letting me choose which webcam to choose. So it just chooses on its own the internal one and I can't change that.

This was in Windows 10, not 11.
